docs/design/operator.md

```diff
@@ -0,0 +1,200 @@
+# Gluster Operator 1.0 Design
+
+## Introduction
+
+This document specifies the overall design for a Gluster operator. An operator
+is a Kubernetes [controller](https://github.com/kubernetes/sample-controller)
+that is focused on managing a specific application with the goal of automating
+common tasks that an administrator would typically perform.
+
+This document is a formalization of many hours of discussion by several people.
+Credit and thanks to Humble Devassy Chirammal (@humblec),Steve Watt
+(@wattsteve), and John Strunk (@JohnStrunk) for their early and insightful
+contributions to the discussions.
+
+This document aims to settle on a concrete set of functionality that a Gluster
+operator should have. It is outside the scope of this document to specify any
+implementation details of this functionality. The desired functionality should
+not be strictly limited by the extents of current-day technologies but should
+not dismiss the costs of waiting for and executing forward development in those
+technologies.
+
+In addition, this document only specifies the desired functionality for
+declaring a 1.0 release. Further enhancements and feature roadmaps will be
```

Comment:
I'd rather say something along the lines of "...specifies a desired minimal set of functionalities for getting started with an operator" ... (again the 1.0 comment).